library/system.lisp

```diff
@@ -81,57 +72,108 @@ Sleep uses type class `Rational`'s `best-approx` instead of `Real`'s `real-appro
 
 (coalton-toplevel
 
-  (declare get-run-time (Unit -> UFix))
+  (declare get-run-time (Unit -> Integer))
   (define (get-run-time)
-    "Gets the run-time."
-    (lisp UFix ()
+    "Gets the run-time in internal time units. This is implementation specific: it may measure real time, run time, CPU cycles, or some other quantity.
+
+The difference between two successive calls to this function represents quantity accumulated during that period of time."
+    (lisp Integer ()
       (cl:get-internal-run-time)))
 
-  (declare get-real-time (Unit -> UFix))
+  (declare get-real-time (Unit -> Integer))
   (define (get-real-time)
-    "Gets the real-time."
-    (lisp UFix ()
+    "Gets the real-time in internal time units. The difference between two successive calls to this function represents the time that has elapsed."
+    (lisp Integer ()
       (cl:get-internal-real-time)))
 
-  #+sbcl
-  (declare get-bytes-consed (Unit -> UFix))
-  #+sbcl
-  (define (get-bytes-consed)
-    "Gets the number of bytes consed (only implemented for SBCL"
-    (lisp UFix ()
-      (sb-ext:get-bytes-consed)))
+  (declare internal-time-units-per-second Integer)
+  (define  internal-time-units-per-second
+    "The number of internal time units per second. This is implementation specific."
+    (lisp Integer ()
+      cl:internal-time-units-per-second))
+
+  (declare time-units->seconds (Integer -> Fraction))
+  (define (time-units->seconds t)
+    "Converts internal time units into `Fraction` seconds."
+    (math:mkfraction t internal-time-units-per-second))
```

Comment:
there's gotta be a better way than mkfraction :|

Reply:
I tried using `/` and it errored on compile:

```
  read-error: 
    COMMON-LISP:READ error during COMMON-LISP:COMPILE-FILE:
    
      error: Type mismatch
      --> system.lisp:24:2
        |
     24 |     (define (time-units->seconds t)
        |  ___^
     25 | |     "Converts internal time units into `Fraction` seconds."
     26 | |     (math:/ t internal-time-units-per-second))
        | |______________________________________________^ Expected type '(INTEGER → FRACTION)' but got type '(INTEGER → INTEGER)'
```

Reply:
general/

/ is only when the input and output types match

Reply:
or exact/